puppet/zulip/manifests/app_frontend_base.pp

```diff
@@ -70,12 +63,21 @@
   # Realm level.
   $tornado_processes = Integer(zulipconf('application_server', 'tornado_processes', 1))
   if $tornado_processes > 1 {
-    $tornado_ports = range(9800, 9800 + $tornado_processes)
+    $tornado_ports = range(9800, 9800 + $tornado_processes - 1)
     $tornado_multiprocess = true
   } else {
     $tornado_multiprocess = false
   }
 
+  file { '/etc/nginx/zulip-include/upstreams':
+    require => Package[$zulip::common::nginx],
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0644',
+    content => template('zulip/nginx/upstreams.conf.template.erb'),
+    notify  => Service['nginx'],
+  }
```

Comment:
Do we want `notify` or `before` here?  Or maybe both is best??  I think we need this to be in place before we want nginx to be started.

Reply:
the other nginx config files seem to be using `notify`, without `before`, so I assume `before` is redundant :thinking: